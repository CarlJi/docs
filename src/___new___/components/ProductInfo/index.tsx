import * as React from 'react';
import { Flex, Box, SystemStyleObject, Heading, Text, Image } from '@chakra-ui/react';
import compareInfo from '../../data/compareInfo';
import {
  prodContainerStyles,
  sectionCol1Styles,
  sectionCol2Styles,
  sectionInnerStyles,
  sectionOuterStyles,
  sectionScreenImage,
} from './styles';
import { headerTextStyle, heading2Styles, textContentStyles } from '../styles';
import LearnMoreButton from '../LearnMoreButton';

interface ProductInfoProps {
  sx?: SystemStyleObject;
}

const ProductInfo: React.FC<ProductInfoProps> = ({ sx, ...rest }) => {
  return (
    <Flex
      sx={{ ...prodContainerStyles, ...sx }}
      {...rest}
    >
      <Heading
        as='h2'
        size='md'
        sx={{ ...heading2Styles(false), ...headerTextStyle }}
      >
        About Tigera products
      </Heading>
      {compareInfo.aboutInfo.map((info, index) => {
        const hasDarkBg = index % 2 !== 0;

        return (
          <Flex
            sx={sectionOuterStyles(false)}
            key={index}
          >
            <Flex sx={sectionInnerStyles(hasDarkBg)}>
              <Box sx={sectionCol1Styles(hasDarkBg)}>
                <Image
                  src={info.image}
                  sx={sectionScreenImage}
                />
              </Box>
              <Flex sx={sectionCol2Styles(hasDarkBg)}>
                <Heading
                  as='h2'
                  size='md'
                  sx={heading2Styles(false)}
                >
                  {info.title}
                </Heading>
                <Box>
                  {info.description.map((desc, index) => (
                    <Text
                      key={index}
                      sx={textContentStyles(false)}
                    >
                      {desc}
                    </Text>
                  ))}
                </Box>

                <Flex justifyContent={['center', 'center', 'start']}>
                  <LearnMoreButton
                    href={info.link}
                    ariaLabel={info.linkDescription}
                    hasDarkBg={false}
                  />
                </Flex>
              </Flex>
            </Flex>
          </Flex>
        );
      })}
    </Flex>
  );
};

export default ProductInfo;
