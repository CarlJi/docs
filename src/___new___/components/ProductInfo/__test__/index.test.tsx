import { render } from '@testing-library/react';

import ProductInfo from '../index';

describe('<ProductInfo/>', () => {
  it('should render the ProductInfo component', () => {
    const { asFragment } = render(<ProductInfo />);
    expect(asFragment()).toMatchSnapshot();
  });
});
