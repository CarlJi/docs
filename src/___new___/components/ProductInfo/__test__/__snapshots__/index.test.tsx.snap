// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductInfo/> should render the ProductInfo component 1`] = `
<DocumentFragment>
  <div
    class="css-1050wj9"
  >
    <h2
      class="chakra-heading css-pxkard"
    >
      About Tigera products
    </h2>
    <div
      class="css-rbv4x0"
    >
      <div
        class="css-1cbf54u"
      >
        <div
          class="css-sg0obw"
        >
          <img
            class="chakra-image css-4lmifj"
            src="img/calicoOpenPI.png"
          />
        </div>
        <div
          class="css-k7jigo"
        >
          <h2
            class="chakra-heading css-7jrbve"
          >
            Calico Open Source
          </h2>
          <div
            class="css-0"
          >
            <p
              class="chakra-text css-1y5m0qd"
            >
              The base product that comprises both Calico Enterprise and Calico Cloud. It provides the core networking and network policy features.
            </p>
          </div>
          <div
            class="css-pay0va"
          >
            <a
              aria-label="Learn more about Calico Open Source"
              class="chakra-link chakra-button css-dh8svx"
              href="/calico/latest/about/"
            >
              Go to docs
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-rbv4x0"
    >
      <div
        class="css-8i50ao"
      >
        <div
          class="css-1irrp4u"
        >
          <img
            class="chakra-image css-4lmifj"
            src="img/calicoEntreprisePI.png"
          />
        </div>
        <div
          class="css-1h9v7dl"
        >
          <h2
            class="chakra-heading css-7jrbve"
          >
            Calico Enterprise
          </h2>
          <div
            class="css-0"
          >
            <p
              class="chakra-text css-1y5m0qd"
            >
              Includes Calico Open Source core networking and network policy, but adds advanced features for networking, network policy, visibility and troubleshooting, threat defense, and compliance reports. 
            </p>
          </div>
          <div
            class="css-pay0va"
          >
            <a
              aria-label="Learn more about Enterprise"
              class="chakra-link chakra-button css-dh8svx"
              href="/calico-enterprise/latest/about-calico-enterprise"
            >
              Go to docs
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-rbv4x0"
    >
      <div
        class="css-1cbf54u"
      >
        <div
          class="css-sg0obw"
        >
          <img
            class="chakra-image css-4lmifj"
            src="img/calicoCloudPI.png"
          />
        </div>
        <div
          class="css-k7jigo"
        >
          <h2
            class="chakra-heading css-7jrbve"
          >
            Calico Cloud
          </h2>
          <div
            class="css-0"
          >
            <p
              class="chakra-text css-1y5m0qd"
            >
              The SaaS version of Calico Enterprise. It adds Image Assurance to scan and detect vulnerabilities in images, and container threat defense to detect malware. It also adds onboarding tutorials, and eliminates the cost to manage Elasticsearch logs and storage that comes with Calico Enterprise.
            </p>
          </div>
          <div
            class="css-pay0va"
          >
            <a
              aria-label="Learn more about Calico Cloud"
              class="chakra-link chakra-button css-dh8svx"
              href="calico-cloud"
            >
              Go to docs
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
